go/shared.go

```diff
@@ -0,0 +1,60 @@
+package app
+
+import (
+	"go.temporal.io/sdk/workflow"
+)
+
+const TripPlanningTaskQueue = "TRIP_PLANNING_TASK_QUEUE"
+
+// Distinct from golang's Time because we just want year and month
+type CreditCardExpiration struct {
+	Year  int
+	Month int
+}
+
+type CreditCardInfo struct {
+	Number     int
+	Ccv        int
+	Expiration CreditCardExpiration
+}
+
+type BookingInfo struct {
+	Name    string
+	Address string
+	CcInfo  CreditCardInfo
+}
+
+type Compensations struct {
+	compensations []any
+	arguments     [][]any
+}
+
+func (s *Compensations) AddCompensation(activity any, parameters ...any) {
+	s.compensations = append(s.compensations, activity)
+	s.arguments = append(s.arguments, parameters)
+}
+
+func (s Compensations) Compensate(ctx workflow.Context, inParallel bool) {
+	if !inParallel {
+		for i := len(s.compensations) - 1; i >= 0; i-- {
```

Comment:
Add comment for why we're going LIFO for the !parallel compensations?

Reply:
Done.